src/pages/HabitTracker/Habits.tsx

```diff
@@ -0,0 +1,224 @@
+  margin-left: 10px;
+  width: 100%;
+  box-sizing: border-box;
+  overflow-x: hidden;
+`;
+
+// Types
+interface Habit {
+  name: string;
+  count: number;
+  index: number;
+}
+
+// Functional component for Habits
+export function Habits() {
+  const [habits, setHabits] = useState<Habit[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [newHabitName, setNewHabitName] = useState("");
+  const [newHabitCount, setNewHabitCount] = useState("");
+  const [editIndex, setEditIndex] = useState<number | null>(null);
+  const [originalHabitName, setOriginalHabitName] = useState("");
+  const [originalHabitCount, setOriginalHabitCount] = useState("");
+
+  // Load habits from local storage when the component mounts
+  useEffect(() => {
+    const storedHabits = localStorage.getItem("habits");
+    if (storedHabits) {
+      setHabits(JSON.parse(storedHabits));
+    }
+  }, []);
+
+  // Handler for showing the add habit modal
+  const handleAddHabitClick = () => {
+    setShowModal(true);
+    resetForm();
+  };
+
+  // Reset the form fields and edit state
+  const resetForm = () => {
+    setNewHabitName("");
+    setNewHabitCount("");
+    setEditIndex(null);
+    setOriginalHabitName("");
+    setOriginalHabitCount("");
+  };
+
+  // Handler for saving a new or edited habit
+  const handleSaveHabit = () => {
+    if (newHabitName && newHabitCount) {
+      const updatedHabits = [...habits];
+      // Update the habit if editing
+      if (editIndex !== null) {
+        updatedHabits[editIndex] = {
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: editIndex + 1,
+        };
+      } else {
+        // Add the new habit
+        updatedHabits.push({
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: habits.length + 1,
+        });
+      }
+      setHabits(updatedHabits);
+      localStorage.setItem("habits", JSON.stringify(updatedHabits));
+      setShowModal(false);
+      resetForm();
+    }
+  };
+
+  // Handler for closing the modal without saving
+  const handleCloseModal = () => {
+    setShowModal(false);
+    resetForm();
+  };
+
+  // Handler for deleting a habit
+  const handleDeleteHabit = (index: number) => {
+    // Filter out the deleted habit and reindex the remaining habits
+    const updatedHabits = habits
+      .filter((habit) => habit.index !== index)
+      .map((habit, idx) => ({ ...habit, index: idx + 1 }));
+    setHabits(updatedHabits);
+    localStorage.setItem("habits", JSON.stringify(updatedHabits));
+  };
+
+  // Handler for editing a habit
+  const handleEditHabit = (index: number) => {
+    const habitToEdit = habits.find((habit) => habit.index === index);
+    if (habitToEdit) {
+      setNewHabitName(habitToEdit.name);
+      setNewHabitCount(habitToEdit.count.toString());
+      setOriginalHabitName(habitToEdit.name);
+      setOriginalHabitCount(habitToEdit.count.toString());
+      setEditIndex(index - 1);
```

Comment:
I am not sure why do we have to update the index here.
Actually, a better name for `index` is `id` in your case.
Index is something we use for referencing the array. `ID` is what is used to identify a record inside an array.